pwem/protocols/protocol_sets.py

```diff
@@ -884,3 +884,84 @@ def _summary(self):
                        ' particles.' % (self.outputParticles.getSize(),
                                         self.inputParticles.get().getSize())]
         return summary
+
+
+class ProtSubSetRange(ProtSets):
+    """
+    Create a subset from a set based on a range.
+
+    This overlaps with more general protocol subset 'range' option, but this one
+    is more specific and much faster.
+    """
+    _label = 'subset range'
+
+    # -------------------------- DEFINE param functions -----------------------
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+
+        add = form.addParam  # short notation
+        add('inputSet', pwprot.params.PointerParam, pointerClass='EMSet',
+            label="Input set of items", important=True,
+            help='Input from where the subset will be created.')
+        add('rangeFrom', pwprot.params.IntParam, default=1,
```

Comment:
I like more the idea of the String param and specifying the id like 1-200, 250-300 ...

It is more flexible...not sure if this is precisely the origin of the performance issue in the "traditional subsets" protocol?